third_party/tsl/third_party/gpus/cuda/hermetic/cuda_redist_init_repositories.bzl

```diff
@@ -218,6 +218,10 @@ def _create_libcuda_symlinks(
         repository_ctx.symlink(nvidia_driver_path, "lib/libcuda.so.1")
         repository_ctx.symlink("lib/libcuda.so.1", "lib/libcuda.so")
 
+def _create_cuda_header_symlinks(repository_ctx):
+    if repository_ctx.name == "cuda_nvcc":
+        repository_ctx.symlink("../cuda_cudart/include/cuda.h", "include/cuda.h")
```

Comment:
Would you explain the algorithm here please, e.g. how does it work?
When does Clang use this symlink?

Reply:
Clang uses `cuda.h` in a location relative to nvcc to determine the cuda version. If it is missing, Clang assumes the latest known Cuda version which is usually incorrect. The hermetic toolchain breaks this by downloading nvcc and cudart to separate directories.

See: https://github.com/llvm/llvm-project/blob/b592917eec407a01ef07334af5aafe6c8922c80a/clang/lib/Driver/ToolChains/Cuda.cpp#L212

Reply:
Thank you. Are you planning to use this change for CLANG builds (not CLANG+NVCC)?

Reply:
Yes, this is for CLANG builds.

Reply:
I think this change merely exposes an already broken build configuration. In your case you should set `-Wno-unknown-cuda-version` to get the same behavior as today. Defaulting to `CudaVersion::NEW` completely ignores the hermetic nvcc version and causes isa generation incompatible with ptxas.

Reply:
In this case may I ask you to change .bazelrc file and add the following to the `build:cuda_clang` config:

`build:cuda_clang --copt="-Wno-unknown-cuda-version"`

And then I'll be able to approve it.